kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/checks/ChannelChecks.kt

```diff
@@ -2,130 +2,211 @@
 
 package com.kotlindiscord.kord.extensions.checks
 
-import com.gitlab.kordlib.core.entity.channel.Category
-import com.gitlab.kordlib.core.entity.channel.Channel
-import com.gitlab.kordlib.core.event.message.MessageCreateEvent
+import com.gitlab.kordlib.core.behavior.channel.CategoryBehavior
+import com.gitlab.kordlib.core.behavior.channel.ChannelBehavior
+import com.gitlab.kordlib.core.event.Event
 import kotlinx.coroutines.flow.toList
+import mu.KotlinLogging
 
 /**
- * Check asserting that a [MessageCreateEvent] fired within a given channel.
+ * Check asserting that an [Event] fired within a given channel.
+ *
+ * Only events that can reasonably be associated with a single channel are supported. Please raise
+ * an issue if an event you expected to be supported, isn't.
  *
  * @param channel The channel to compare to.
  */
-fun inChannel(channel: Channel): suspend (MessageCreateEvent) -> Boolean {
-    suspend fun inner(event: MessageCreateEvent): Boolean {
-        with(event) {
-            return message.channel.id == channel.id
+fun inChannel(channel: ChannelBehavior): suspend (Event) -> Boolean {
+    val logger = KotlinLogging.logger {}
+
+    suspend fun inner(event: Event): Boolean {
+        val eventChannel = channelFor(event)
+
+        if (eventChannel == null) {
+            logger.debug { "Channel for event $event is null. This type of event may not be supported." }
```

Comment:
I feel like this should be a warning instead, or at least an info, in case debug level loggings are disabled, this can save some troubleshooting.

Reply:
This is true, but the main reason I did it is that this message really only matters to the developer. Additionally, it's going to get noisy - there are lots of _valid_ event types that will also return `null` for these function calls.

Reply:
Ah I see the issue now, debug level seems fair.